CorsixTH/Lua/dialogs/build_room.lua

```diff
@@ -102,16 +102,19 @@ function UIBuildRoom:UIBuildRoom(ui)
   self:makeTooltip(_S.tooltip.build_room_window.cost, 160, 228, 282, 242)
 end
 
+--! Checks what rooms are now available to build
 function UIBuildRoom:updateBuildableRooms()
   local app = self.ui.app
   self.category_rooms = {}
   for i, category in ipairs({"diagnosis", "treatment", "clinics", "facilities"}) do
     local rooms = {}
     self.category_rooms[i] = rooms
-    for _, room in ipairs(app.world.available_rooms) do
+    local room_discoveries = self.ui.hospital.room_discoveries
+    for room_id, _ in pairs(room_discoveries) do
+      local room = room_discoveries[room_id].room
       -- NB: Unimplemented rooms are hidden unless in debug mode
       if (app.config.debug or room.class) and room.categories[category] and
-          self.ui.hospital.discovered_rooms[room] then
+          room_discoveries[room_id].is_discovered then
```

Comment:
The 2nd result of `pairs` at line 113 is the value of `room_discoveries[room_id]`. I am not sure why you discard that value at line 113, and then query it twice at lines 114 and 117